op-service/sources/l1_beacon_client.go

```diff
@@ -63,6 +80,106 @@ func (cl *L1BeaconClient) apiReq(ctx context.Context, dest any, reqPath string,
+	var configResp eth.APIConfigResponse
+	if err := cl.apiReq(ctx, &configResp, specMethod, nil); err != nil {
+		return eth.APIConfigResponse{}, err
+	}
+	return configResp, nil
+}
+
+func (cl *BeaconHTTPClient) BeaconGenesis(ctx context.Context) (eth.APIGenesisResponse, error) {
+	var genesisResp eth.APIGenesisResponse
+	if err := cl.apiReq(ctx, &genesisResp, genesisMethod, nil); err != nil {
+		return eth.APIGenesisResponse{}, err
+	}
+	return genesisResp, nil
+}
+
+func (cl *BeaconHTTPClient) BeaconBlobSideCars(ctx context.Context, fetchAllSidecars bool, slot uint64, hashes []eth.IndexedBlobHash) (eth.APIGetBlobSidecarsResponse, error) {
+	reqPath := path.Join(sidecarsMethodPrefix, strconv.FormatUint(slot, 10))
+	var reqQuery url.Values
+	if !fetchAllSidecars {
+		reqQuery = url.Values{}
+		for i := range hashes {
+			reqQuery.Add("indices", strconv.FormatUint(hashes[i].Index, 10))
+		}
+	}
+	var resp eth.APIGetBlobSidecarsResponse
+	if err := cl.apiReq(ctx, &resp, reqPath, reqQuery); err != nil {
+		return eth.APIGetBlobSidecarsResponse{}, err
+	}
+	return resp, nil
+}
+
+// FallbackBeaconClient implents the BeaconClient interface. It has two underlying BeaconClients,
+// it first queries the primary and then queries the secondary if the first call fails
+type FallbackBeaconClient struct {
+	primary   BeaconClient
+	secondary BeaconClient
+	l         log.Logger
+}
+
+func NewFallbackClient(a, b client.HTTP, l log.Logger) *FallbackBeaconClient {
+	return &FallbackBeaconClient{
+		primary:   &BeaconHTTPClient{cl: a},
+		secondary: &BeaconHTTPClient{cl: b},
+		l:         l,
+	}
+}
+
+func (cl *FallbackBeaconClient) NodeVersion(ctx context.Context) (string, error) {
+	if r, e := cl.primary.NodeVersion(ctx); e == nil {
+		return r, nil
+	} else {
+		cl.l.Warn("failed to get node version from primary", "err", e)
+	}
+	return cl.secondary.NodeVersion(ctx)
+}
+
+func (cl *FallbackBeaconClient) ConfigSpec(ctx context.Context) (eth.APIConfigResponse, error) {
+	if r, e := cl.primary.ConfigSpec(ctx); e == nil {
+		return r, nil
+	} else {
+		cl.l.Warn("failed to get config from primary", "err", e)
+	}
+	return cl.secondary.ConfigSpec(ctx)
+}
+
+func (cl *FallbackBeaconClient) BeaconGenesis(ctx context.Context) (eth.APIGenesisResponse, error) {
+	if r, e := cl.primary.BeaconGenesis(ctx); e == nil {
+		return r, nil
+	} else {
+		cl.l.Warn("failed to get beacon genesis from primary", "err", e)
+	}
+	return cl.secondary.BeaconGenesis(ctx)
+}
+
+func (cl *FallbackBeaconClient) BeaconBlobSideCars(ctx context.Context, fetchAllSidecars bool, slot uint64, hashes []eth.IndexedBlobHash) (eth.APIGetBlobSidecarsResponse, error) {
+	if r, e := cl.primary.BeaconBlobSideCars(ctx, fetchAllSidecars, slot, hashes); e == nil {
+		return r, nil
+	} else if strings.Contains(e.Error(), "404") {
+		// Log at debug if 404 because this is expected to happen when fetching old blobs.
+		cl.l.Debug("failed to get beacon blob sidecars from primary", "err", e)
```

Comment:
maybe use a different log message so it's easier to differentiate from the warning in `else`
```suggestion
		// Log at debug if 404 because this is expected to happen when fetching old blobs.
		cl.l.Debug("blob sidecars not found on primary beacon", "err", e)
```